subsys/bluetooth/host/id.c

```diff
@@ -864,13 +864,19 @@ void bt_id_add(struct bt_keys *keys)
 		goto done;
 	}
 
+	if (duplicate_rl &&  !IS_ENABLED(CONFIG_BT_PRIVACY)) {
```

Comment:
What about the rl_entries == rl_size branch above? That would still increment rl_entries even for a duplicate. 

Also (minor issue): you have two spaces after &&

Reply:
The way I see it, rl_entries is used to check if there's any address resolution.  In this situation, it should have been counted before.  So there's no need to increment.